src/provider/provider_level_zero.c

```diff
@@ -114,6 +130,13 @@ umf_result_t ze_memory_provider_initialize(void *params, void **provider) {
     ze_provider->device = ze_params->level_zero_device_handle;
     ze_provider->memory_type = (ze_memory_type_t)ze_params->memory_type;
 
+    ze_result_t ret = g_ze_ops.zeDeviceGetProperties(
+        ze_provider->device, &ze_provider->device_properties);
+    if (ret != ZE_RESULT_SUCCESS) {
+        umf_ba_global_free(ze_provider);
+        return UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC;
+    }
+
```

Comment:
could you add checks for resident_device_count and resident_device_handles? I would expect that the number of not-NULL resident_device_handles should match the count

Reply:
I'm not sure if this is needed. If we call zeContextMakeMemoryResident we will get INVALID argument anyway in that case. I think we should just create a mapping from L0 error codes to UMF error codes.

Reply:
Done.